addons/toolbars/src/components/ToolbarManager.tsx

```diff
@@ -1,36 +1,34 @@
 import React, { FC } from 'react';
 import { useGlobalTypes } from '@storybook/api';
 import { Separator } from '@storybook/components';
-
+import { ToolbarMenuCycle } from './ToolbarMenuCycle';
+import { ToolbarMenuList } from './ToolbarMenuList';
+import { normalizeArgType } from '../utils/normalize-toolbar-arg-type';
 import { ToolbarArgType } from '../types';
-import { MenuToolbar } from './MenuToolbar';
-
-const normalize = (key: string, argType: ToolbarArgType) => ({
-  ...argType,
-  name: argType.name || key,
-  description: argType.description || key,
-  toolbar: {
-    ...argType.toolbar,
-    items: argType.toolbar.items.map((item) =>
-      typeof item === 'string' ? { value: item, title: item } : item
-    ),
-  },
-});
 
 /**
  * A smart component for handling manager-preview interactions.
  */
 export const ToolbarManager: FC = () => {
   const globalTypes = useGlobalTypes();
-  const keys = Object.keys(globalTypes).filter((key) => !!globalTypes[key].toolbar);
-  if (!keys.length) return null;
+  const globalIds = Object.keys(globalTypes).filter((id) => !!globalTypes[id].toolbar);
+
+  if (!globalIds.length) {
+    return null;
+  }
 
   return (
     <>
       <Separator />
-      {keys.map((key) => {
-        const normalizedConfig = normalize(key, globalTypes[key] as ToolbarArgType);
-        return <MenuToolbar key={key} id={key} {...normalizedConfig} />;
+      {globalIds.map((id) => {
+        const normalizedArgType = normalizeArgType(id, globalTypes[id] as ToolbarArgType);
+        const isCycle = normalizedArgType.toolbar.cycle === true;
```

Comment:
What's use case for not having `cycle`? Can we simplify this?

Reply:
it's horribly named - it should be `toggle` - but toggle indicates only being able to toggle between two states - and it can handle x states

It invokes the no menu and button only for cycling between item values